buildlib/azure-pipelines-pr.yml

```diff
@@ -252,19 +252,64 @@ stages:
               set -eE
               . buildlib/tools/common.sh
               prepare_build
-              ${WORKSPACE}/configure $(CONFIGURE_OPTS)
+              ${WORKSPACE}/configure $(CONFIGURE_OPTS) --prefix=$(Build.Repository.LocalPath)/install
               gcc -v
               make -s -j `nproc`
-              set +eE
-              set -x
-              ./src/tools/info/ucx_info -e -u t 2>&1 | tee info.txt
+              make install
+            displayName: Build ucx artifact
+          - task: CopyFiles@2
+            inputs:
+              sourceFolder: '$(Build.Repository.LocalPath)'
+              contents: |
+                install/**
+              targetFolder: '$(Build.ArtifactStagingDirectory)'
+          - task: PublishBuildArtifacts@1
+            inputs:
+              pathToPublish: '$(Build.ArtifactStagingDirectory)'
+              artifactName: ucx_bin_$(Build.BuildId)
+      - job: Test_in_containers
+        dependsOn: Distros
+        displayName: UCX tests in containers
+        pool:
+          name: MLNX
+          demands:
+          - ucx_docker -equals yes
+        strategy:
+          matrix:
+            ubuntu20:
+              CONTAINER: ubuntu2004
+        container: $[ variables['CONTAINER'] ]
+        steps:
+          - checkout: none
+          - task: DownloadBuildArtifacts@0
+            displayName: 'Download Build Artifacts'
+            inputs:
+              artifactName: ucx_bin_$(Build.BuildId)
+              downloadPath: $(System.DefaultWorkingDirectory)
+          - bash: chmod u+rwx $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId) -R
+          - bash: |
+              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
+              cd $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install
+          - bash: |
+              set +eE -x
+              ./bin/ucx_info -e -u t 2>&1 | tee info.txt
               grep -i error info.txt
               retVal=$?
               if [ $retVal -eq 0 ]; then
                   exit 1;
               fi
               exit 0;
             displayName: Test ucx_info
+          - bash: |
+              set +eE -x
+              ./bin/ucx_perftest -t tag_bw &
+              server_pid=$!
+              sleep 1;
+              ./bin/ucx_perftest -t tag_bw localhost 2>&1 | tee perf.txt
+              wait $server_pid
+              grep "Final:" perf.txt
+              exit $?;
```

Comment:
why needed?

Reply:
i mean, if we just do "set -Eex", i'd expect grep failure will lead to step failed. no need also have exit command